libs/core/src/main/java/org/opensearch/common/io/stream/Writeable.java

```diff
@@ -5,73 +5,101 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.common.io.stream;
 
 import java.io.IOException;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
 /**
- * Implementers can be written to a {@code StreamOutput} and read from a {@code StreamInput}. This allows them to be "thrown
+ * Implementers can be written to a {@linkplain StreamOutput} and read from a {@linkplain StreamInput}. This allows them to be "thrown
  * across the wire" using OpenSearch's internal protocol. If the implementer also implements equals and hashCode then a copy made by
  * serializing and deserializing must be equal and have the same hashCode. It isn't required that such a copy be entirely unchanged.
  *
  * @opensearch.internal
  */
-public interface BaseWriteable<S extends BaseStreamOutput> {
+public interface Writeable {
     /**
      * A WriteableRegistry registers {@link Writer} methods for writing data types over a
-     * {@link BaseStreamOutput} channel and {@link Reader} methods for reading data from a
-     * {@link BaseStreamInput} channel.
+     * {@link StreamOutput} channel and {@link Reader} methods for reading data from a
+     * {@link StreamInput} channel.
      *
      * @opensearch.internal
      */
     class WriteableRegistry {
-        private static final Map<Class<?>, Writer<? extends BaseStreamOutput, ?>> WRITER_REGISTRY = new ConcurrentHashMap<>();
+        private static final Map<Class<?>, Writer<?>> WRITER_REGISTRY = new ConcurrentHashMap<>();
         private static final Map<Class<?>, Class<?>> WRITER_CUSTOM_CLASS_MAP = new ConcurrentHashMap<>();
-        private static final Map<Byte, Reader<? extends BaseStreamInput, ?>> READER_REGISTRY = new ConcurrentHashMap<>();
+        private static final Map<Byte, Reader<?>> READER_REGISTRY = new ConcurrentHashMap<>();
 
         /**
          * registers a streamable writer
          *
          * @opensearch.internal
          */
-        public static <W extends Writer<? extends BaseStreamOutput, ?>> void registerWriter(final Class<?> clazz, final W writer) {
+        public static <W extends Writer<?>> void registerWriter(final Class<?> clazz, final W writer) {
             if (WRITER_REGISTRY.putIfAbsent(clazz, writer) != null) {
                 throw new IllegalArgumentException("Streamable writer already registered for type [" + clazz.getName() + "]");
             }
+            WRITER_REGISTRY.put(clazz, writer);
```

Comment:
unnecessary extra put here and online 80.

Reply:
There was another one I took out as well.